extensions/interactions/MusicNotesInput/directives/music-notes-input-rules.service.ts

```diff
@@ -16,81 +16,108 @@
+    var answerLength:number = (
+      MusicNotesInputRulesService._convertSequenceToMidi(answer).length);
+    return answerLength >= inputs.a && answerLength <= inputs.b;
+  }
+  IsEqualToExceptFor(answer: Note[], inputs: {x: Note[], k: number}): boolean {
+    var targetSequence: number[] = (
+      MusicNotesInputRulesService._convertSequenceToMidi(inputs.x));
+    var userSequence: number[] = (
+      MusicNotesInputRulesService._convertSequenceToMidi(answer));
+    if (userSequence.length !== targetSequence.length) {
+      return false;
+    }
 
-        var numWrongNotes = 0;
-        userSequence.map(function(noteValue, index) {
-          if (noteValue !== targetSequence[index]) {
-            numWrongNotes++;
-          }
-        });
-        return numWrongNotes <= inputs.k;
-      },
-      IsTranspositionOf: function(answer, inputs) {
-        var targetSequence = _convertSequenceToMidi(inputs.x);
-        var userSequence = _convertSequenceToMidi(answer);
-        if (userSequence.length !== targetSequence.length) {
-          return false;
-        }
-        return userSequence.every(function(noteValue, index) {
-          return targetSequence[index] + inputs.y === noteValue;
-        });
-      },
-      IsTranspositionOfExceptFor: function(answer, inputs) {
-        var targetSequence = _convertSequenceToMidi(inputs.x);
-        var userSequence = _convertSequenceToMidi(answer);
-        if (userSequence.length !== targetSequence.length) {
-          return false;
-        }
+    var numWrongNotes: number = 0;
+    userSequence.map(function(noteValue, index) {
+      if (noteValue !== targetSequence[index]) {
+        numWrongNotes++;
+      }
+    });
+    return numWrongNotes <= inputs.k;
+  }
+  IsTranspositionOf(answer: Note[], inputs: {x: Note[], y: number}): boolean {
+    var targetSequence: number[] = (
+      MusicNotesInputRulesService._convertSequenceToMidi(inputs.x));
+    var userSequence: number[] = (
+      MusicNotesInputRulesService._convertSequenceToMidi(answer));
+    if (userSequence.length !== targetSequence.length) {
+      return false;
+    }
+    return userSequence.every((noteValue, index) => {
+      return targetSequence[index] + inputs.y === noteValue;
+    });
+  }
+  IsTranspositionOfExceptFor(
+      answer: Note[], inputs: {x: Note[], y: number, k: number}): boolean {
+    var targetSequence: number[] = (
+      MusicNotesInputRulesService._convertSequenceToMidi(inputs.x));
+    var userSequence: number[] = (
+      MusicNotesInputRulesService._convertSequenceToMidi(answer));
+    if (userSequence.length !== targetSequence.length) {
+      return false;
+    }
 
-        var numWrongNotes = 0;
-        userSequence.map(function(noteValue, index) {
-          if (targetSequence[index] + inputs.y !== noteValue) {
-            numWrongNotes++;
-          }
-        });
-        return numWrongNotes <= inputs.k;
+    var numWrongNotes: number = 0;
+    userSequence.map((noteValue, index) => {
```

Comment:
Do we need `function` keyword as it is or are we changing this to `fat function (arrow function)`?

I'm not sure which of these will work with Angular 8, my guess is both would work. Though, I haven't yet figured out the exact path.

I've somewhat gone through the style guide: https://angular.io/guide/styleguide#member-sequence. Here you can see `window.settimout()` is used by arrow function. There is an example upgrading the phone service -- I couldn't see use of `function` keyword anywhere here too.  This leads me to think we should go on with arrow function.

It also might be the case `function` and `arrow functions` both could work with Angular 8, and it could be the difference in ES. ES6 supports use of arrow function whereas ES5 does not https://engineering.carsguide.com.au/es5-vs-es6-syntax-6c8350fa6998 and I guess we can use ES5 and ES6 together (like mix of two).

cc/ @seanlip @kevinlee12 @YashJipkate

Reply:
I believe either is fine. I think this is indeed more of an ES difference rather than an Angular difference.

@kevinlee12 any advice?

Reply:
I think we should go with arrow functions.

Reply:
I agree with Vojta on this one.

Reply:
Noted, thanks @seanlip, @vojtechjelinek, @kevinlee12.

I'll go with the arrow functions (though I'm still reading about the Angular, AngularJS, ES5 and ES6 and will let you know if I find anything else).